ext/webserial/01_webserial.js

```diff
@@ -0,0 +1,397 @@
+      }
+    }
+
+    get writable() {
+      webidl.assertBranded(this, SerialPort);
+      if (this[_writable] !== null) {
+        return this[_writable];
+      }
+      if (this[_state] !== "opened") {
+        return null;
+      }
+      if (this[_writeFatal]) {
+        return null;
+      }
+
+      const stream = new WritableStream({
+        write: async (chunk) => {
+          const source = webidl.converters.BufferSource(chunk);
+          await core.opAsync("op_webserial_write", this[_rid], source.slice()); // TODO(@crowlKats): errors
+        },
+        abort: () => {
+          // TODO(@crowlKats): Invoke the operating system to discard the contents of all software and hardware transmit buffers for the port.
+          this.#handleClosingWritable();
+        },
+        close: () => {
+          // TODO(@crowlKats): Invoke the operating system to flush the contents of all software and hardware transmit buffers for the port.
+          this.#handleClosingWritable();
+        },
+      }, {
+        highWaterMark: this[_bufferSize],
+        size: (chunk) => chunk.byteLength,
+      });
+
+      this[_writable] = stream;
+      return stream;
+    }
+
+    #handleClosingWritable() {
+      this[_writable] = null;
+      if (this[_readable] === null && this[_pendingClosePromise] !== null) {
+        this[_pendingClosePromise].resolve(undefined);
+      }
+    }
+
+    constructor() {
+      webidl.illegalConstructor();
+    }
+
+    getInfo() {
+      webidl.assertBranded(this, SerialPort);
+      return this[_info];
+    }
+
+    // deno-lint-ignore require-await
+    async open(options) {
+      webidl.assertBranded(this, SerialPort);
+      const prefix = "Failed to execute 'open' on 'SerialPort'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
+      options = webidl.converters.SerialOptions(options, {
+        prefix,
+        context: "Argument 1",
+      });
+
+      if (this[_state] !== "closed") {
+        throw new DOMException(
+          "SerialPort must be closed",
+          "InvalidStateError",
+        );
+      }
+
+      if (options.dataBits !== 7 || options.dataBits !== 8) {
+        throw new TypeError("Invalid 'dataBits' given, must be either 7 or 8.");
+      }
+      if (options.stopBits !== 1 || options.stopBits !== 2) {
+        throw new TypeError("Invalid 'stopBits' given, must be either 1 or 2.");
+      }
+      if (options.bufferSize === 0) {
+        throw new TypeError(
+          "Invalid 'bufferSize' given, must be greater than 0.",
+        );
+      }
+
+      this[_state] = "opening";
+      this[_rid] = core.opSync("op_webserial_open_port", this[_name], options);
+      this[_state] = "opened";
+      this[_bufferSize] = options.bufferSize;
+    }
+
+    async setSignals(options = {}) {
+      webidl.assertBranded(this, SerialPort);
+      const prefix = "Failed to execute 'setSignals' on 'SerialPort'";
+      webidl.requiredArguments(arguments.length, 1, { prefix });
```

Comment:
This seems like an odd sort of assert. I guess it would only fire if there are more than 1 arguments passed to the function? I guess WebIDL is like that.

Reply:
technically not required.